nixos/modules/programs/slock.nix

```
@@ -16,16 +16,17 @@ in
           Whether to install slock screen locker with setuid wrapper.
         '';
       };
+      package = mkPackageOption pkgs "slock" {};
     };
   };
 
   config = mkIf cfg.enable {
-    environment.systemPackages = [ pkgs.slock ];
+    environment.systemPackages = [ cfg.package ];
     security.wrappers.slock =
       { setuid = true;
         owner = "root";
         group = "root";
-        source = "${pkgs.slock.out}/bin/slock";
+        source = "${cfg.package.out}/bin/slock";
```

Might be worth using `lib.getExe` here.
```suggestion
        source = "${lib.getExe cfg.package.out}";
```

---

Also fair, but needs `meta.mainProgram` to be defined in slock's recipe (it's not)